### DISCUSSION TITLE
How to get router instance in setup store?

### DISCUSSION THREAD

**Discussion by jsormaz** [vuejs/pinia#1954] [Help and Questions] [answered]
I'm trying to use a setup store so I can make use of watch()

however now I can't get access to the router.

with options stores, `this.router.push()` works.

with setup stores i tried 
`import { useRouter } from "vue-router"; 
const router = useRouter();`
 but router is still undefined?

**Comment by posva** [comment, marked as answer]
if you are not doing SSR, just import the `router` instance from where it's declared. If you are using SSR, you will have to wait for 

- https://github.com/vuejs/pinia/issues/1784
